torch_geometric/nn/aggr/base.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import torch
+from torch import Tensor
+from torch_scatter import scatter, segment_csr
+
+
+class BaseAggr(torch.nn.Module, ABC):
+    r"""An abstract base class for writing aggregations."""
+    @abstractmethod
+    def forward(self, x: Tensor, index: Tensor, dim_size: Optional[int] = None,
+                ptr: Optional[Tensor] = None, dim: int = -2) -> Tensor:
+        pass
```

Comment:
would be a great opportunity to write a nice docstring about what each of these are in case someone wants to subclass it themselves

Reply:
done :)